httpserver/httpserver.go

```diff
@@ -23,22 +24,27 @@ type Server struct {
 
 type ServerOption func(*Server)
 
-// WithHandler adds a handler on the http endpoint.
-func WithHandler(h HandleChainFunc) ServerOption {
+// WithHandler adds a handler on the http endpoint. Nil argument is ignored.
+func WithHandler(handler HandleChainFunc) ServerOption {
 	return func(s *Server) {
-		if h != nil {
-			s.handlers = append(s.handlers, h)
+		if handler != nil && !reflect.ValueOf(handler).IsNil() {
+			s.handlers = append(s.handlers, handler)
 		}
 	}
 }
 
-// WithHandlers adds a handler on the http endpoint.
-func WithHandlers(h []HandleChainFunc) ServerOption {
+// WithHandlers adds a handler on the http endpoint. Nil arguments is ignored.
```

Comment:
its good that we are checking for nil!

Do we want to ignore?

If i am trying to register a handler on the http endpoint, I would like a good indication if it has successfully been added? I am not a fan of silently failing. What case would we pass nil to WithHandlers?

Reply:
this comment applies to all the other spots we are ignoring nil as well

Reply:
its a deliberate design choice to avoid unnecssary checks in the main code. The API explicitly states that nil values are ignored and if they dont start because you have made a mistake you will detect it.

Reply:
all fixed